homeassistant/components/dhcp/__init__.py

```diff
@@ -317,14 +318,16 @@ async def async_start(self) -> None:
             self._async_process_device_state(state)
 
     @callback
-    def _async_process_device_event(self, event: Event) -> None:
+    def _async_process_device_event(
+        self, event: EventType[EventStateChangedData]
+    ) -> None:
         """Process a device tracker state change event."""
         self._async_process_device_state(event.data["new_state"])
 
     @callback
-    def _async_process_device_state(self, state: State) -> None:
+    def _async_process_device_state(self, state: State | None) -> None:
         """Process a device tracker state."""
-        if state.state != STATE_HOME:
+        if state is None or state.state != STATE_HOME:
```

Comment:
I guess this was a bug? But not likely to get very far if it's none anyways

Reply:
Not sure it was really a bug. Technically the method is only called for `async_track_state_added_domain` with the value of `new_state`, so I would probably expect it to always be `State`. However just from a typing perspective it's probably easier to reuse `EventStateChangedData` instead of creating something new as it's always called after a `EVENT_STATE_CHANGED` event.

Maybe it would be possible to do something like #97124, just not sure this is necessary here. It's easy enough to check for `None` values.